src/bopforge/categorical_pipeline/__main__.py

```diff
@@ -0,0 +1,265 @@
+import os
+import shutil
+import warnings
+from argparse import ArgumentParser
+from pathlib import Path
+
+import bopforge.categorical_pipeline.functions as functions
+import numpy as np
+from bopforge.utils import fill_dict, ParseKwargs
+from pplkit.data.interface import DataInterface
+
+warnings.filterwarnings("ignore")
+
+def pre_processing(result_folder: Path) -> None:
+    dataif = DataInterface(result=result_folder)
+    name = dataif.result.name
+
+    # load data
+    df = dataif.load_result(f"raw-{name}.csv")
+    all_settings = dataif.load_result("settings.yaml")
+    settings = all_settings["select_bias_covs"]["cov_finder"]
+
+    # get bias covariates that need to be removed
+    all_covs = [col for col in df.columns if col.startswith("cov_")]
+    covs_to_remove = [col for col in all_covs if len(df[col].unique()) == 1]
```

Comment:
here we probably can use
```python
df[col].unique() == 1
```
But we can make this chance along with removing columns with only several 1s or 0s. And also in other types of pipelines